### ISSUE TITLE
3.x: Single incorrect or missing images (tracking issue)

### ISSUE THREAD

**Issue by akarnokd** [ReactiveX/RxJava#5788]
- [x] `amb`: missing diagram
- [x] `ambArray`: missing diagram
- [x] `concat(arg)`: missing diagram
- [x] `concat(s1..s4)`: add 3rd and 4th source, indicate backpressure
- [x] `concatArray`: missing diagram
- [x] `concatArrayEager`: missing diagram
- [x] `concatEager`: missing diagram
- [x] `create`: missing diagram
- [x] `defer`: missing diagram
- [x] `error(Callable)`: missing diagram
- [x] `error(Throwable)`: indicate error in the box
- [x] `fromCallable`: missing diagram
- [x] `fromPublisher`: missing diagram
- [x] `fromObservable`: missing diagram
- [x] `merge`: missing diagram
- [x] `merge(SingleSource<SingleSource<T>>)`: indicate main element is a `SingleSource`
- [x] `merge(s1..s4)`: add 3rd and 4th source, indicate backpressure
- [x] `mergeDelayError`: Iterable/Publisher missing diagrams, S2,S3,S4 wrong diagrams
- [x] `never`: missing diagram
- [x] `timer`: missing diagram
- [x] `equals`: missing diagram
- [x] `unsafeCreate`: missing diagram
- [x] `using`: missing diagram
- [x] `wrap`: missing diagram
- [x] `ambWith`: missing diagram
- [x] `as`: missing diagram
- [x] `hide`: missing diagram
- [x] `compose`: missing diagram
- [x] `cache`: missing diagram
- [x] `cast`: missing diagram
- [x] `delay`: missing diagram
- [x] `delaySubscription`: missing diagram
- [x] `dematerialize`: missing diagram
- [x] `doAfterSuccess`: missing diagram
- [x] `doAfterTerminate`: indicate error case
- [x] `doFinally`: missing diagram
- [x] `doOnSubscibe`: missing diagram
- [x] `doOnSuccess`: missing diagram
- [x] `doOnEvent`: missing diagram
- [x] `doOnError`: missing diagram
- [x] `doOnDispose`: missing diagram
- [x] `filter`: one source item, indicate both true and false outcomes
- [x] `flatMapPublisher`: operator name
- [x] `blockingGet`: missing diagram
- [x] `lift`: missing diagram
- [x] `contains`: missing diagram
- [x] `mergeWith`: indicate backpressure
- [x] `onErrorReturn`: indicate item is the result of a function call with the original error
- [x] `onErrorReturnItem`: use the previous `onErrorReturn` diagram, change name in box
- [x] `onErrorResumeNext`: diagram should have one item and no complete marker
- [x] `onErrorResumeWith`: operator name
- [x] `onTerminateDetach`: missing diagram
- [x] `repeat`: missing diagram
- [x] `repeatWhen`: missing diagram
- [x] `repeatUntil`: missing diagram
- [x] `retry`: missing diagram
- [x] `retryWhen`: missing diagram
- [x] `subscribe`: missing diagram
- [x] `subscribeWith`: missing diagram?
- [x] `takeUntil`: fix item count in main and other sources, indicate all other signal types
- [x] `timeout`: missing diagram
- [x] `to`: missing diagram
- [x] `toCompletable`: diagram is stretched
- [x] `toFlowable`: operator name in box, indicate backpressure
- [x] `toFuture`: indicate Future, remove the blocking text
- [x] `toMaybe`: operator name in box, indicate both outcomes with proper signal types
- [x] `unsubscribeOn`: missing diagram
- [x] `test`: missing diagram?

**Comment by VeskoI** [comment]
I see most marbles in the JavaDoc live under this folder:  `https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/`. Are other locations accepted - e.g. can I upload it in Google Drive for example and open a PR with the updated path in the JavaDoc directly? Here's a few I created:

![single_dofinally](https://user-images.githubusercontent.com/4089934/39448132-dca93836-4cbb-11e8-9b13-ce078a8ba206.png)
![single_doondispose](https://user-images.githubusercontent.com/4089934/39448133-dcca516a-4cbb-11e8-9143-b935cf0dd64c.png)
![single_doonerror](https://user-images.githubusercontent.com/4089934/39448135-dce99bce-4cbb-11e8-9d52-8e30084f9aa4.png)
![single_doonsubscribe](https://user-images.githubusercontent.com/4089934/39448136-dd05bcbe-4cbb-11e8-823d-d9d0c3501f24.png)
![single_doonsuccess](https://user-images.githubusercontent.com/4089934/39448137-dd21fdca-4cbb-11e8-848d-45c3af91b76f.png)

**Comment by akarnokd** [comment]
Images have to be hosted by the project. Besides, Google drive tends to change the service terms and images may become inaccessible (I've lost resource location this way in another project). 

About your images:

- `doFinally`: 
  - error should be a red X
  - the dispose case puts the star too close to the time arrow as if it was an output instead of a side-effect
  - I draw disposed flows with dashed lines ([example](https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/doFinally.o.png)), it would be great here
- `doOnDispose`
  - the dispose case puts the star too close to the time arrow as if it was an output instead of a side-effect
  - I draw disposed flows with dashed lines ([example](https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/doFinally.o.png)), it would be great here
  - `unsubscribe` should be `dispose`
- `doOnSubscribe`
  - the star should be in between the operator box and the bottom timeline
- `doOnSuccess`: OK

**Comment by akarnokd** [comment]
Updated images from #5987 now in the repository:

![doFinally](https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doFinally.png)
![doOnSuccess](https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnSuccess.2.png)
![doOnError](https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnError.2.png)
![doOnDispose](https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnDispose.png)
![doOnSubscribe](https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnSubscribe.png)

**Comment by AarjavP** [comment]
how are these made? are there any written guidelines? I can try making some when I have time.

**Comment by akarnokd** [comment]
There is a Mac only OmniGraffle. I'm on Windows and I use Excel shapes and arrows.

**Comment by VeskoI** [comment]
@AarjavP , I used [THIS template](https://github.com/davidmoten/rxjava-marble-template) as a staring point. It's for **Inkscape** software, which worked nicely on my Mac.

**Comment by UMFsimke** [comment]
Hi all, can you please tell me is this marble acceptable for the `filter` operator?

As I haven't contributed to this project so far I would like to take it easy and slow :) Let me know what should I change so that I have a reference for future marbles.

![filter](https://user-images.githubusercontent.com/10116857/42415703-8d0530ce-8257-11e8-966c-2756a336e6c1.png)

**Comment by akarnokd** [comment]
Not really. There is no | in Single. Please have a look at diagrams above: https://github.com/ReactiveX/RxJava/issues/5788#issuecomment-385533074

**Comment by UMFsimke** [comment]
@akarnokd can we update this issue please? Its getting a bit messy and hard to follow what has been completed.

**Comment by akarnokd** [comment]
@UMFsimke It has been only you lately that worked on this.
